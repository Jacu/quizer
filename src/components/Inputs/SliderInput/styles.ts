import styled from 'styled-components';
import * as colors from '~/utils/Colors';

export const SliderInput = styled.div`
    display: flex;
    margin: 10px 0px;
`;

export const ValueLabel = styled.div`
    font-size: 1.5rem;
    flex: 1;
`;

export const Label = styled.div`
    width: 100%;
    flex: 1;
    text-align: end;
    font-size: 1.5rem;
    padding: 0 20px;
    border-right: 1px solid ${colors.gray.base};
`;

export const SliderContainer = styled.div`
    width: 100%;
    flex: 4;
    display: flex;
    align-items: center;
`;

export const Slider = styled.div`
    flex: 3;
    padding-right: 50px;
`;