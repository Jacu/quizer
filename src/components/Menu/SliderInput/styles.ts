import styled from 'styled-components';

export const SliderInput = styled.div`
    display: flex;
`;

export const ValueLabel = styled.div`
    font-size: 1.5rem;
    flex: 1;
`;

export const Label = styled.div`
    width: 100%;
    flex: 1;
    text-align: end;
`;

export const SliderContainer = styled.div`
    width: 100%;
    flex: 4;
    display: flex;
    align-items: center;
`;

export const Slider = styled.div`
    flex: 2;
`;