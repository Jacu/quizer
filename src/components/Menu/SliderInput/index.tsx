import React from 'react';
import * as styled from './styles';
import Slider from '~/components/UI/Slider/Slider';

interface ISliderInput {
    label: string,
    min: number,
    max: number,
    value: string,
    onChange: (newValue: string) => void,
}

const SliderInput: React.FC<ISliderInput> = (props) => {
    return (
        <styled.SliderInput>
            <styled.Label>{props.label}</styled.Label>
            <styled.SliderContainer>
                <styled.ValueLabel>{props.value}</styled.ValueLabel>
                <styled.Slider>
                    <Slider {...props} />
                </styled.Slider>
            </styled.SliderContainer>
        </styled.SliderInput>
    )
};

export default SliderInput;