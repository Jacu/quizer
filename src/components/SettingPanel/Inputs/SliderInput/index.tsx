import React from 'react';
import * as styled from './styles';
import Slider from '~/components/UI/Slider';
import { IInput } from '../';

const SliderInput: React.FC<IInput<number>> = ({ label, selected, options, onChange }) => {
    return (
        <styled.SliderInput>
            <styled.Label>{label}</styled.Label>
            <styled.SliderContainer>
                <styled.ValueLabel>{selected}</styled.ValueLabel>
                <styled.Slider>
                    <Slider value={selected} min={options[0] || 0} max={options[1] || 1} onChange={onChange} />
                </styled.Slider>
            </styled.SliderContainer>
        </styled.SliderInput>
    )
};

export default SliderInput;