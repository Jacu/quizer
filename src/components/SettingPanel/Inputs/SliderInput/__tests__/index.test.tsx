import React from 'react';
import SliderInput from '../';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

const onChange = jest.fn();

const props = {
    label: 'sliderInput',
    options: [1, 20],
    selected: 5,
    onChange,
};

describe('SliderInput', () => {
    it('should render with default props', () => {
        const { asFragment } = render(<SliderInput {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });

    describe('should read min and max values from props', () => {
        const testOptions = [0, 666]
        it('should set min value for slider input', () => {
            const { container } = render(<SliderInput {...props} options={testOptions} />);
            expect(container.querySelector('input')).toHaveAttribute('min', '0')
        });

        it('should set max value for slider input', () => {
            const { container } = render(<SliderInput {...props} options={testOptions} />);
            expect(container.querySelector('input')).toHaveAttribute('max', '666')
        });

        it('should use default values if none provided', () => {
            const { container } = render(<SliderInput {...props} options={[]} />);
            expect(container.querySelector('input')).toHaveAttribute('min', '0');
            expect(container.querySelector('input')).toHaveAttribute('max', '1');
        });
    });

    it('should use value from props', () => {
        const testValue = 10;
        const { container } = render(<SliderInput {...props} selected={testValue} />);
        expect(container.querySelector('input')).toHaveAttribute('value', testValue.toString());
    });

    it('should call onChange', () => {
        const { container } = render(<SliderInput {...props} />);
        fireEvent.change(container.querySelector('input')!, { target: { value: 10 } });
        expect(onChange).toHaveBeenCalledWith(10);
    });
});