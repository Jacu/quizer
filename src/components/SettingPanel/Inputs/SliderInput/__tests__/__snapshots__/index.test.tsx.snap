// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderInput should render with default props 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm Taep"
  >
    <div
      class="sc-AxiKw hqYDqk"
    >
      sliderInput
    </div>
    <div
      class="sc-AxhCb KZzGZ"
    >
      <div
        class="sc-AxirZ cQnzXZ"
      >
        5
      </div>
      <div
        class="sc-AxhUy igmPPg"
      >
        <input
          class="sc-AxgMl jPPVud"
          max="20"
          min="1"
          step="5"
          type="range"
          value="5"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
